File: Objective-C/ExampleApplication/ExampleApplication/AdFormatViewControllers/InterstitialViewController.m
//
//  InterstitialViewController.m
//  ExampleApplication
//
//  Created by Levent ORAL on 23.11.2018.
//  Copyright © 2018 App Samurai Inc. All rights reserved.
//

#import "InterstitialViewController.h"
@import AppSamuraiAdSDK;

@interface InterstitialViewController ()

@property(nonatomic, strong) ASInterstitial *asInterstitial;

@end

@implementation InterstitialViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    // Initialize ASInterstitial
    _asInterstitial = [[ASInterstitial alloc] initWithAdUnitID:@"appsamurai-sample-ios-interstitial-ad-id"];
    // delegate is used to receive ad events
    _asInterstitial.delegate = self;
    
    // Load ad with request
    [_asInterstitial loadAdWithAdRequest:[[ASAdRequest alloc] init]];
}

- (IBAction)onShowInterstitialAdClicked:(id)sender {
    // Show ad when isReady is true and hasBeenUsed is false. so you can only show ad when it's ready and has not used before
    if (_asInterstitial != nil && _asInterstitial.isReady && !_asInterstitial.hasBeenUsed) {
        // Present ad
        [_asInterstitial presentWithRootViewController:self];
    }
}

#pragma ASRewardBasedVideoAdDelegate functions
// Notify when ad is succesfully received
- (void)interstitialDidReceiveAd:(ASInterstitial *_Nonnull)asInterstitial {
    NSLog(@"%s", "interstitialDidReceiveAd");
}

// Notify when ad is failed to receive, check ASAdRequestError types and console logs for details
- (void)interstitialDidFailToReceiveAd:(ASInterstitial *_Nonnull)asInterstitial error:(ASAdRequestError *_Nonnull)error {
    NSLog(@"%s", "interstitialDidFailToReceiveAd");
}

// Notify when ad will be presented screen on fullscreen
- (void)interstitialWillPresentScreen:(ASInterstitial *_Nonnull)asInterstitial {
    NSLog(@"%s", "interstitialWillPresentScreen");
}

// Notify when ad will dismiss from screen
- (void)interstitialWillDismissScreen:(ASInterstitial *_Nonnull)asInterstitial {
    NSLog(@"%s", "interstitialWillDismissScreen");
}

// Notify when ad dismissed from screen
- (void)interstitialDidDismissScreen:(ASInterstitial *_Nonnull)asInterstitial {
    NSLog(@"%s", "interstitialDidDismissScreen");
}

// Notify when ad is clicked and another app will be opened
- (void)interstitialWillLeaveApplication:(ASInterstitial *_Nonnull)asInterstitial {
    NSLog(@"%s", "interstitialWillLeaveApplication");
}

@end

File: Swift/ExampleApplication/ExampleApplication/AdFormatViewControllers/InterstitialViewController.swift
//
//  InterstitialViewController.swift
//  ExampleApplication
//
//  Created by Levent ORAL on 23.11.2018.
//  Copyright © 2018 App Samurai Inc. All rights reserved.
//

import UIKit
import AppSamuraiAdSDK

class InterstitialViewController: UIViewController, ASInterstitialDelegate {
    
    private var asInterstitial: ASInterstitial?

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
        
        // Initialize ASInterstitial
        asInterstitial = ASInterstitial(adUnitID: "appsamurai-sample-ios-interstitial-ad-id")
        // Initialize ASInterstitial with optional gadAdUnitID for those who want to use Google Ad Mob flow
//        asInterstitial = ASInterstitial(adUnitID: "appsamurai-sample-ios-interstitial-ad-id", gadAdUnitID: "/6499/example/interstitial")
        // delegate is used to receive ad events
        asInterstitial?.delegate = self
        
        // Load ad with request
        asInterstitial?.loadAd(adRequest: ASAdRequest())
    }
    
    @IBAction func onShowInterstitialAdClicked(_ sender: Any) {
        // Show ad when isReady is true and hasBeenUsed is false. so you can only show ad when it's ready and has not used before
        if (asInterstitial?.isReady ?? false) && !(asInterstitial?.hasBeenUsed ?? true) {
            // Present ad
            asInterstitial?.present(rootViewController: self)
        }
    }
    
    //MARK: - ASInterstitialDelegate functions
    // Notify when ad is succesfully received
    func interstitialDidReceiveAd(_ asInterstitial: ASInterstitial) {
        print("interstitialDidReceiveAd")
    }
    
    // Notify when ad is failed to receive, check ASAdRequestError types and console logs for details
    func interstitialDidFailToReceiveAd(_ asInterstitial: ASInterstitial, error: ASAdRequestError) {
        print("interstitialDidFailToReceiveAd \(error.localizedDescription)")
    }
    
    // Notify when ad will be presented screen on fullscreen
    func interstitialWillPresentScreen(_ asInterstitial: ASInterstitial) {
        print("interstitialWillPresentScreen")
    }
    
    // Notify when ad will dismiss from screen
    func interstitialWillDismissScreen(_ asInterstitial: ASInterstitial) {
        print("interstitialWillDismissScreen")
    }
    
    // Notify when ad dismissed from screen
    func interstitialDidDismissScreen(_ asInterstitial: ASInterstitial) {
        print("interstitialDidDismissScreen")
    }
    
    // Notify when ad is clicked and another app will be opened
    func interstitialWillLeaveApplication(_ asInterstitial: ASInterstitial) {
        print("interstitialWillLeaveApplication")
    }
}
